From cd0fcfe605df6efbfc51aadb19f840b4e3239bc5 Mon Sep 17 00:00:00 2001
From: Rob Johnston <rob.johnston@joyent.com>
Date: Tue, 4 Jun 2019 15:24:56 -0700
Subject: [PATCH] TRITON-1651 sdc-usbkey: set-variable should enclose value in
 quotes Reviewed by: John Levon <john.levon@joyent.com> Reviewed by: Trent
 Mick <trent.mick@joyent.com> Approved by: Trent Mick <trent.mick@joyent.com>

---
 tools/lib/usbkey.js | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/tools/lib/usbkey.js b/tools/lib/usbkey.js
index 6c46adef..04a89aa9 100644
--- a/tools/lib/usbkey.js
+++ b/tools/lib/usbkey.js
@@ -1193,6 +1193,13 @@ set_variable_loader(mountpoint, name, value, callback)
     mod_assert.func(callback, 'callback');
 
     var search = '^\\s*' + name + '\\s*=\\s*.*$';
+
+    /*
+     * Make sure the value is enclosed in quotes (") in case the value contains
+     * embedded whitespace.  Otherwise, loader will fail to parse it correctly.
+     */
+    value = '"' + value + '"';
+
     var replace = name + '=' + value;
 
     sedfile(mountpoint + '/boot/loader.conf', search, replace, callback);
-- 
2.21.0

